test/unit/Parser.spec.ts

```diff
@@ -0,0 +1,153 @@
+    it(`should parse ${packet.type}`, (done) => {
+      verify([packet])
+        .then(done)
+        .catch(done);
+
+      const packetStr = packet.toRaw();
+      parser.feed(packetStr + delimiter);
+    });
+  }
+
+  function testSplitPacket(packet: Packet) {
+    it(`should parse ${packet.type} split`, (done) => {
+      verify([packet])
+        .then(done)
+        .catch(done);
+
+      const packetStr = packet.toRaw();
+      const middleIndex = packetStr.length >> 1;
+      parser.feed(packetStr.slice(0, middleIndex));
+      parser.feed(packetStr.slice(middleIndex) + delimiter);
+    });
+  }
+
+  function testConcatenatedPackets(packets: Packet[]) {
+    it(`should parse ${packets.map(packet => packet.type).join(' ')} concatenated`, (done) => {
+      verify(packets)
+        .then(done)
+        .catch(done);
+
+      parser.feed(packets.map(packet => packet.toRaw() + delimiter).join(''));
+    });
+  }
+
+  function testConcatenatedAndSplitOnTheDelimiter(packets: Packet[]) {
+    it(`should parse ${packets.map(packet => packet.type).join(' ')} concatenated and split on the delimiter`, (done) => {
+      verify(packets)
+        .then(done)
+        .catch(done);
+
+      let remaining = '';
+      packets.forEach((packet) => {
+        const packetStr = remaining + packet.toRaw() + delimiter;
+        const chunk = packetStr.slice(0, packetStr.length - 1); // split just before the delimiter ends
+        remaining = packetStr.slice(packetStr.length - 1); // keep the remaining of the delimiter for the next chunk
+        parser.feed(chunk);
+      });
+      parser.feed(remaining);
+    });
+  }
+
+  const pingPacket = new PingPacket();
+  const helloPacket = new HelloPacket({
+    version: '1.0.0',
+    nodePubKey: '0479BE667EF9DCBBAC55A06295CE870B07029BFCDB2DCE28D959F2815B16F8179',
+    listenPort: 4001,
+    pairs: ['BTC/LTC'],
+    raidenAddress: '8483ADA7726A3C4655DA4FBFC0E1108A8FD17B448A68554199C47D08FFB10D4B8',
+  });
+
+  testPacket(pingPacket);
+  testPacket(helloPacket);
+  testSplitPacket(pingPacket);
+  testSplitPacket(helloPacket);
+  testConcatenatedPackets([pingPacket, helloPacket, pingPacket]);
+  testConcatenatedAndSplitOnTheDelimiter([pingPacket, helloPacket, pingPacket]);
+
+  it(`should not try to parse an empty string`, (done) => {
+    wait()
+      .then(() => expect.fail())
+      .catch((err) => {
+        expect(err).to.equal('timeout');
+        done();
+      });
+
+    parser.feed('');
+  });
+
+  it(`should try parse a standalone delimiter and fail`, (done) => {
+    wait()
+      .then(() => expect.fail())
+      .catch((err) => {
+        expect(err.type).to.equal(ParserErrorType.UNPARSEABLE_MESSAGE);
+        done();
+      });
+
+    parser.feed(delimiter);
+  });
+
+  it(`should buffer a max buffer length`, (done) => {
+    parser = new Parser(delimiter, 10);
+    wait()
+      .then(() => expect.fail())
+      .catch((err) => {
+        expect(err).to.equal('timeout');
```

Comment:
I'm getting an unhandled rejection error from this line not every time I run the tests locally but most times. If you merge master, I'm using the `chai-as-promised` plugin for some of the sanity tests. Maybe you can use it here as well with the sanity tests as an example.

Reply:
The 'timeout' exception is expected. If you're talking about tests timeout (20 sec), then it doesn't happen for me locally.
I just saw your `chai-as-promised` usage - looks nice.